Apex/Assignment 2/src/classes/TrainingSessionManager.cls

```diff
@@ -0,0 +1,287 @@
+/**Purpose:(Assignment 2) Class To Perform Create, Read, Update, Delete on Training_Session__c.
+
+Created by: Mukul Pichunia
+
+Created Date : 27-11-24
+
+Rivision Log: V:1.1
+
+**/
+public without sharing class TrainingSessionManager {
+    public Enum SessionsType {
+    WEBINAR, WORKSHOP, SEMINAR
+	}
+    public Enum Status {
+        SUCCESS, FAILED
+    }
+    //Constructor
+    public TrainingSessionManager() {
+
+    }
+
+    /*
+     * @description: Creates a new Training_Session__c record.
+     * @param sessionName The name of the training session.
+     * @param typeOfSession The type of session (WEBINAR, WORKSHOP, SEMINAR).
+     * @param sessionDate The date of the session.
+     * @param location The location of the session (ignored for webinars).
+     * @return The created Training_Session__c record.
+     * @throws IllegalArgumentException if sessionName is blank or sessionDate is in the past.
+     */
+    public Training_Session__c createTrainingSessions(String sessionName,SessionsType typeOfSession,  Date sessionDate, String location) {
+        // Check for sessionName validity
+        if (String.isBlank(sessionName)) {
+            // Throw an exception if sessionName is blank
+            throw new IllegalArgumentException('Session Name must not be empty.');
+        }
+        if (sessionDate < Date.today()) {
+            // Throw an exception if sessionDate is in past
+            throw new IllegalArgumentException('Session Date must not be in past');
+        }
+        // Create new Training_Session__c record
+        Training_Session__c trainingSessions = new Training_Session__c();
+        trainingSessions.name = sessionName;
+        trainingSessions.Session_Date__c = sessionDate;
+        // Use of Enum
+        if (typeOfSession == TrainingSessionManager.SessionsType.WEBINAR) {
```

Comment:
Try the code after removing class name.

Reply:
Sure

Reply:
Yes We can Do it After removing the class name it is working as Expected